src/unique_path.cpp

```diff
@@ -111,7 +111,9 @@ void system_crypt_random(void* buf, std::size_t len, boost::system::error_code*
 
   if (!errval)
   {
-    BOOL gen_ok = ::CryptGenRandom(handle, len, static_cast<unsigned char*>(buf));
+    // Use static cast to silence Visual Studio 2017
+    // warning C4267: 'argument': conversion from 'size_t' to 'DWORD', possible loss of data
+    BOOL gen_ok = ::CryptGenRandom(handle, static_cast<DWORD>(len), static_cast<unsigned char*>(buf));
```

Comment:
@Lastique here's another random implementation that is not ready for Windows UWP.

Reply:
Should we be using `static_cast` or should we be using `boost::numeric_cast` to ensure proper handling of math here (and the other instances I see in the other file).

Reply:
numeric_cast seems appropriate. I will make that change.